components/org.wso2.carbon.identity.organization.management.service/src/main/java/org/wso2/carbon/identity/organization/management/service/dao/impl/CacheBackedOrganizationManagementDAO.java

```diff
@@ -0,0 +1,358 @@
+
+        return organizationMgtDAO.getOrganization(organizationId);
+    }
+
+    @Override
+    public List<BasicOrganization> getOrganizations(boolean recursive, Integer limit, String organizationId,
+                                                    String sortOrder, List<ExpressionNode> expressionNodes,
+                                                    List<ExpressionNode> parentIdExpressionNodes)
+            throws OrganizationManagementServerException {
+
+        return organizationMgtDAO.getOrganizations(recursive, limit, organizationId, sortOrder, expressionNodes,
+                parentIdExpressionNodes);
+    }
+
+    @Override
+    public void deleteOrganization(String organizationId) throws OrganizationManagementServerException {
+
+        String tenantDomain = resolveTenantDomain(organizationId);
+        organizationMgtDAO.deleteOrganization(organizationId);
+        if (tenantDomain != null) {
+            clearOrganizationDetailsCache(organizationId, tenantDomain);
+        }
+        clearTenantDomainCache(organizationId);
+    }
+
+    @Override
+    public boolean hasChildOrganizations(String organizationId) throws OrganizationManagementServerException {
+
+        return organizationMgtDAO.hasChildOrganizations(organizationId);
+    }
+
+    @Override
+    public void patchOrganization(String organizationId, Instant lastModifiedInstant,
+                                  List<PatchOperation> patchOperations) throws OrganizationManagementServerException {
+
+        String tenantDomain = resolveTenantDomain(organizationId);
+        organizationMgtDAO.patchOrganization(organizationId, lastModifiedInstant, patchOperations);
+        if (tenantDomain != null) {
+            clearOrganizationDetailsCache(organizationId, tenantDomain);
+        }
+    }
+
+    @Override
+    public void updateOrganization(String organizationId, Organization organization)
+            throws OrganizationManagementServerException {
+
+        String tenantDomain = resolveTenantDomain(organizationId);
+        organizationMgtDAO.updateOrganization(organizationId, organization);
+        if (tenantDomain != null) {
+            clearOrganizationDetailsCache(organizationId, tenantDomain);
+        }
+    }
+
+    @Override
+    public boolean isAttributeExistByKey(String organizationId, String attributeKey)
+            throws OrganizationManagementServerException {
+
+        return organizationMgtDAO.isAttributeExistByKey(organizationId, attributeKey);
+    }
+
+    @Override
+    public List<BasicOrganization> getChildOrganizations(String organizationId, boolean recursive)
+            throws OrganizationManagementServerException {
+
+        return organizationMgtDAO.getChildOrganizations(organizationId, recursive);
+    }
+
+    @Override
+    public List<String> getChildOrganizationIds(String organizationId) throws OrganizationManagementServerException {
+        return null;
```

Comment:
missing a new line after the method signature. And returning null is wrong, Need to invoke DAO method